runtime-light/stdlib/array/array-functions.h

```diff
@@ -32,13 +53,51 @@ void f$shuffle(array<T> &arr) noexcept {
 }
 
 template<class T>
-array<array<T>> f$array_chunk(const array<T> &a, int64_t chunk_size, bool preserve_keys = false) {
-  php_critical_error("call to unsupported function");
+task_t<array<T>> f$array_filter(const array<T> &a) noexcept {
+  co_return co_await array_functions_impl_::array_filter_impl(a, [](const auto &it) { return f$boolval(it.get_value()); });
+}
+
+template<class T, class Callback>
+task_t<array<T>> f$array_filter(const array<T> &a, const Callback &callback) noexcept {
+  if constexpr (is_async_function_v<Callback, T>) {
+    co_return co_await array_functions_impl_::array_filter_impl(a, [&callback](const auto &it) -> task_t<bool> {
+      co_return f$boolval(co_await callback(it.get_value()));
+    });
+  } else {
+    co_return co_await array_functions_impl_::array_filter_impl(a, [&callback](const auto &it) { return f$boolval(callback(it.get_value())); });
+  }
 }
 
 template<class T>
-array<T> f$array_slice(const array<T> &a, int64_t offset, const mixed &length_var = mixed(), bool preserve_keys = false) {
-  php_critical_error("call to unsupported function");
+typename array<T>::key_type f$array_rand(const array<T> &a) noexcept {
+  if (int64_t size = a.count()) {
+    return a.middle(f$mt_rand(0, size - 1)).get_key();
+  }
+  return {};
+}
+
+template<class T>
+mixed f$array_rand(const array<T> &a, int64_t num) noexcept {
+  if (num == 1) {
+    return f$array_rand(a);
+  }
+
+  int64_t size = a.count();
+
+  if (unlikely(num <= 0 || num > size)) {
```

Comment:
As light runtime is written in C++-20, we have an access to standard mechanisms like `[[likely]]`